iidm/iidm-serde/src/main/java/com/powsybl/iidm/serde/extensions/BranchStatusSerDe.java

```diff
@@ -12,30 +12,30 @@
 import com.powsybl.commons.io.DeserializerContext;
 import com.powsybl.commons.io.SerializerContext;
 import com.powsybl.iidm.network.Connectable;
-import com.powsybl.iidm.network.extensions.BranchStatus;
-import com.powsybl.iidm.network.extensions.BranchStatusAdder;
+import com.powsybl.iidm.network.extensions.OperatingStatus;
+import com.powsybl.iidm.network.extensions.OperatingStatusAdder;
 
 /**
  * @author Nicolas Noir {@literal <nicolas.noir at rte-france.com>}
  */
 @AutoService(ExtensionSerDe.class)
-public class BranchStatusSerDe<C extends Connectable<C>> extends AbstractExtensionSerDe<C, BranchStatus<C>> {
+public class BranchStatusSerDe<C extends Connectable<C>> extends AbstractExtensionSerDe<C, OperatingStatus<C>> {
 
     public BranchStatusSerDe() {
-        super(BranchStatus.NAME, "network", BranchStatus.class,
+        super("branchStatus", "network", OperatingStatus.class,
                 "branchStatus.xsd", "http://www.powsybl.org/schema/iidm/ext/branch_status/1_0",
                 "bs");
     }
 
     @Override
-    public void write(BranchStatus<C> branchStatus, SerializerContext context) {
-        context.getWriter().writeNodeContent(branchStatus.getStatus().name());
+    public void write(OperatingStatus<C> branchStatus, SerializerContext context) {
+        throw new UnsupportedOperationException("This is a deprecated extension (replaced by `OperatingStatus`) and it should be never written anymore");
     }
 
     @Override
-    public BranchStatus<C> read(C connectable, DeserializerContext context) {
-        BranchStatus.Status status = BranchStatus.Status.valueOf(context.getReader().readContent());
-        BranchStatusAdder<C> adder = connectable.newExtension(BranchStatusAdder.class);
+    public OperatingStatus<C> read(C connectable, DeserializerContext context) {
```

Comment:
It might be useful to add a "deprecated" warning/tag so that the users see when reading a file that the extension is deprecated ?

Reply:
This is not really a deprecation as it will stay forever for backward compatibility if XIIDM. I added an explanation in the class Javadoc.